File: Lab9/Lab9_4/Lab9_4/MatrixGenerator.cpp
#include "MatrixGenerator.h"

MatrixGenerator::MatrixGenerator(int size, double maxValue) :MatrixGenerator(size, size, maxValue)
{
}

MatrixGenerator::MatrixGenerator(int rowSize, int columnSize, double maxValue) : rowSize(rowSize), columnSize(columnSize), maxValue(maxValue)
{
}

double** MatrixGenerator::generate()
{
	double** matrix = new double* [rowSize];

	for (int i = 0; i < rowSize; i++) {
		matrix[i] = new double[columnSize];
		for (int j = 0; j < columnSize; j++) {
			std::random_device rd;
			std::mt19937 gen(rd());
			std::uniform_real_distribution<> unif(0.0, maxValue);
			matrix[i][j] = unif(gen);
		}
	}
	return matrix;
}

File: Lab9/Lab9_1/src/main/java/MatrixGenerator.java
import java.util.Random;

public class MatrixGenerator {
    private int rowSize;
    private int columnSize;
    private double maxValue;

    Random random;

    MatrixGenerator(int size, double maxValue) {
        this(size, size, maxValue);
    }

    MatrixGenerator(int rowSize, int columnSize, double maxValue) {
        this.rowSize = rowSize;
        this.columnSize = columnSize;
        this.maxValue = maxValue;

        random = new Random();
    }

    double[][] generate() {
        double[][] matrix = new double[rowSize][columnSize];

        for (int i = 0; i < rowSize; i++) {
            matrix[i] = new double[columnSize];
            for (int j = 0; j < columnSize; j++) {
                matrix[i][j] = maxValue * random.nextDouble();
            }
        }

        return matrix;
    }
}
